qualtran/bloqs/arithmetic/subtraction.py

```diff
@@ -61,7 +63,9 @@ class Subtract(Bloq):
         b: A b_dtype.bitsize-sized input/output register (register b above).
     """
 
-    a_dtype: Union[QInt, QUInt, QMontgomeryUInt] = field()
+    a_dtype: Union[QInt, QUInt, QMontgomeryUInt] = field(
+        converter=lambda k: QUInt(k) if isinstance(k, int) else k
+    )
```

Comment:
let's keep this explicit for now. (if we were to do this we'd want it on both types)

Reply:
done